src/Microsoft.Health.Fhir.Api/Features/Audit/AuditLoggingFilterAttribute.cs

```diff
@@ -49,13 +49,25 @@ public override void OnResultExecuted(ResultExecutedContext context)
 
             Debug.Assert(actionDescriptor != null, "The ActionDescriptor must be ControllerActionDescriptor.");
 
-            var fhirResult = context.Result as FhirResult;
+            // The result can either be a FhirResult or an OperationOutcomeResult which both extend BaseActionResult.
+
+            string typeName = null;
+
+            switch (context.Result)
+            {
+                case FhirResult fhirResult:
```

Comment:
Both FhirResult and OperationOutcomeResult implement BaseActionResult<T> so maybe have BaseActionResult<T> implement an interface that returns the result type so that if we derive another type from the BaseActionResult in the future, we don't need to update the code again?

Reply:
I think I would change the LogExecuted method and rename the parameter from resourceType to responseObjectType and just have the base class implement an interface that returns the Result?.GetType().Name or something. There isn't any restriction that the logging has to be a FHIR resource.